src/pages/Jobs/AllJobs/index.jsx

```diff
@@ -16,20 +17,179 @@
  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
 
-import React from "react";
+import React, { useEffect, useState } from "react";
 
 // Title
 import Title from "components/Title";
 
+// Required functions for API and Error handling
+import { handleError } from "shared/helper";
+import { initialMessage, entriesOptions } from "constants/constants";
+import messages from "constants/messages";
+import { InputContainer, Alert } from "components/Widgets";
+import { getAllJob } from "services/jobs";
+
 const AllJobs = () => {
+  const initialState = {
+    page: 1,
+    limit: 10,
+  };
+  // Setting the browse data to the table
+  const [jobsDataList, setJobsDataList] = useState();
+  const [jobsData, setJobsData] = useState(initialState);
+  // variable for searching in the datatable
+  const [query, setQuery] = useState("");
+
+  // State Variables for handling Error Boundaries
+  const [message, setMessage] = useState(initialMessage);
+  const [showMessage, setShowMessage] = useState(false);
+  const [pagesOptions, setPagesOptions] = useState();
+
+  function Dateformatter(date) {
+    const options = {
+      weekday: "long",
+      year: "numeric",
+      month: "long",
+      day: "numeric",
+    };
+    const d = new Date(date);
+    return d.toLocaleDateString("en-US", options);
+  }
+
+  useEffect(() => {
+    setMessage({
+      type: "success",
+      text: messages.loading,
+    });
+    setShowMessage(true);
+    getAllJob(jobsData)
+      .then((res) => {
+        // Formatting the date from time stamp to readable date
+        setShowMessage(false);
+        const arr = [];
+        for (let i = 0; i < res.res.length; i++) {
+          res.res[i].queueDate = Dateformatter(res.res[i].queueDate);
+          arr[i] = res.res[i];
+        }
+        // Creating pagination
+        const arrPages = [];
+        for (let i = 0; i < res.pages; i++) {
+          arrPages.push({
+            id: i + 1,
+            value: i + 1,
+          });
+        }
```

Comment:
Are you reading the total number of pages from the response header `X-Total-Pages`?
![image](https://user-images.githubusercontent.com/18077542/175230410-86c2bea2-d0d6-4248-b6b5-1cb8fe7668af.png)

Reply:
Yes, the number of pages is fetched from this particular header and then is read from the `res.pages`

Reply:
From this API function:
![Screenshot from 2022-06-23 12-18-34](https://user-images.githubusercontent.com/63705023/175233757-d46c94c8-331b-4bf4-a31a-78c9837ff257.png)
And this services function:
![Screenshot from 2022-06-23 12-18-44](https://user-images.githubusercontent.com/63705023/175233877-e5e5c928-5238-4f3e-8dc7-241d3773d3d2.png)

Reply:
I get you.. but the local storage "pages" is also used in the browse page and is not affecting them. As we are using separate variables and states for the different pages and using useEffect each and every time a page is rendered or state is changed. It should not cause any issue. I did not find any effect of it in other pages. If you do please let me know, will look into the bug then. Thank you for your review!!

Reply:
I see it now, it is parsed in API https://github.com/fossology/FOSSologyUI/blob/183ab4ebb6e98b3856c722d2fe7e1b10e1073992/src/api/sendRequest.js#L85-L97